### PR TITLE
EL-2591

### PR DESCRIPTION
**Pull request by donna-marijne**
Updated to clear the query when closing the search field.
Added a search history which is displayed when the query is empty.
Added an alt+/ keyboard shortcut.

https://jira.autonomy.com/browse/EL-2591

### CODE REVIEW

File: docs/app/components/navigation-bar-search/navigation-bar-search.component.ts [hunk 1]
```diff
@@ -130,28 +146,34 @@ export class NavigationBarSearchComponent {
...
-        }
 
-        // get the latest results
-        this.results = this.data.filter((item: ISearchResult) => {
-            return item.link.title.toLowerCase().indexOf(this.query.toLowerCase()) !== -1;
-        });
+        } else {
+
+            // get the latest results
+            this.results = this.data.filter((item: ISearchResult) => {
```
Comment (ashley-hunter) [inline docs/app/components/navigation-bar-search/navigation-bar-search.component.ts]: I notice if I type in the name of a section with subsections of the same name eg. dashboard, im getting duplicate results, we should probably filter them out?

image

File: docs/app/components/navigation-bar-search/navigation-bar-search.component.ts [hunk 2]
```diff
@@ -160,4 +182,24 @@ export class NavigationBarSearchComponent {
...
         return this.results.filter((result: ISearchResult) => result.link.title === item.link.title).length > 1;
     }
+
+    /**
+     * Add an item to the bounded history list.
+     */
+    private addToHistory(item: ISearchResult) {
+
+        // Remove the item if it's already there.
+        const historyIndex = this.history.indexOf(item);
```
Comment (ashley-hunter) [inline docs/app/components/navigation-bar-search/navigation-bar-search.component.ts]: The old site stored the last 5 results in local storage so they persisted across page refreshes. I realise this was required on the old site as it wasnt a single page application, however do you think this feature should be carried across?

Comment (donna-marijne) [inline docs/app/components/navigation-bar-search/navigation-bar-search.component.ts]: Yes it's a good idea, especially because of the showcase.

**Comment by donna-marijne** [comment]
Made the above changes - thanks.

**Pull request merged**